src/NuGet.Core/NuGet.Logging/CollectorLogger.cs

```diff
@@ -0,0 +1,88 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+
+namespace NuGet.Logging
+{
+    public class CollectorLogger : ICollectorLogger
```

Comment:
Sounds good. Was thinking the next ask would be replay warnings, so I built it this way.